tempy/tempy.py

```diff
@@ -402,6 +406,51 @@ def wrap(self, other):
         other.append(self)
         return self
 
+    def wrap_many(self, *args, strict=False):
+        """Wraps different copies of this element inside all empty tags
+        listed in params or param's (non-empty) iterators.
+
+        Returns list of copies of this element wrapped inside args
+        or None if not succeeded, in the same order and same structure,
+        i.e. args = (Div(), (Div())) -> value = (A(...), (A(...)))
+
+        If on some args it must raise TagError, it will only if strict is True,
+        otherwise it will do nothing with them and return Nones on their positions"""
+        wcopies = []
+        failure = []
+
+        def wrap_next(tag, idx):
+            nonlocal wcopies, failure
+            next_copy = self.__copy__()
+            try:
+                return next_copy.wrap(tag)
+            except TagError:
+                failure.append(idx)
+                return None
```

Comment:
this will end up in `wcopies`, and if strict par is not set to True can lead to some `None` in the return of the main method, i'd skip returning `None` in the `TagError` case, or filtering out None from `wcopies` before the last return.

Reply:
The main idea was that the structure of return values is the same as args
So, if we call `A().wrap_many(Div(), (Div(), Div()), [Div(), Div()])` we get `[A(..), (A(..), A(..)), [A(..), A(..)]]`, so maybe if needed someone can zip theese values into dict or just easily understand where the result for each input is located in output

So, I left theese `None` elements to keep this structure. Maybe it is better to place unwraped copies instead of `None`?

Reply:
Uhm yeah I get your point. By default the single wrap method returns self, so wrap_many should do the same. I'd also return self.__copy__() when TagError is raised.

Reply:
Is there a way to return something if Error was raised? I thought both `return` and `raise` lead to the end of the function.
And am I right that you want this function to return `self` instead of `wcopies` or you just want to replace this `None` with `self.copy()` and still return `wcopies` as the final result?

Reply:
The latter, I think is better to return wcopies as the final result and self.__copy__() in the wrap_next method in the TagError catch.

All my comment here was referred to the wrap_next funciton.

Reply:
Now fixed, thank you :)